lib/features/base/action/ui_action.dart

```diff
@@ -7,11 +7,9 @@ abstract class Action with EquatableMixin {}
 
 abstract class UIAction extends Action {}
 
-abstract class FcmAction extends Action {}
-
-abstract class FcmStateChangeAction extends FcmAction {
+abstract class PushNotificationStateChangeAction extends UIAction {
```

Comment:
It should not inherit from UIAction. It may not need to communicate with the view.

Reply:
Done